mlrun/feature_store/steps.py

```diff
@@ -204,13 +216,45 @@ def _do_pandas(self, event):
             elif feature_map:
                 # create and apply simple map
                 df[self._get_feature_name(feature)] = event[feature].map(
-                    lambda x: feature_map[x]
+                    lambda x: feature_map.get(x, None)
                 )
 
         if self.with_original_features:
             df = pd.concat([event, df], axis=1)
         return df
 
+    def _do_spark(self, event):
+        from itertools import chain
+
+        from pyspark.sql.functions import col, create_map, lit, when
+
+        for column, column_map in self.mapping.items():
+            new_column_name = self._get_feature_name(column)
+            if "ranges" not in column_map:
+                mapping_expr = create_map([lit(x) for x in chain(*column_map.items())])
+                event = event.withColumn(
+                    new_column_name, mapping_expr.getItem(col(column))
```

Comment:
Note that `_get_feature_name` will return the original feature name if `with_original_features` is not set - this means that you're adding a column with a name that already exists in that case -> I'm not sure if Spark allows this or not?

Reply:
Spark overwrite this column, I tested it in `test_ingest_with_steps_mapval` when `with_original_features==False`